### ISSUE TITLE
4.3.5(93/5) implied iterator aspects

### ISSUE THREAD

**Issue by nholsti** [pat-rogers/Ada-202x-WG9-Informal-Review#166]
This example assumes that Map_Type has iterator aspects, which are not shown in its earlier declaration. Perhaps a comment should be added to this effect to avoid confusion.

**Comment by ARG-Editor** [comment]
Agreed. I'm tempted to delete 4.3.5(92/5) and 4.3.5(93/5), even though they are useful examples, because they violate our design principles for examples (in that they be compilable without "magic"). Defining the four needed aspects to support "of" iteration would really hide the purpose of these examples.

The only alternative I can think of would be to somehow depend on the A.18.33 example for these iterators - but I'm not sure how to do that.

**Comment by sttaft** [comment]
At this point I would suggest we do one of the following:
* delete the two bullets; or 
* add a comment saying "--  This presumes the Map type has the necessary aspects to permit iteration -- see A.18.5 for an example of these aspects"; or
* do nothing -- defer to next revision cycle.

I think I favor the last choice.

Interestingly, while looking at A.18.33, I discovered that the instance "Node_Maps" is not used, except in a "use" clause.  This should be remedied some day as well, by changing "Reached_From" into a vector:

      Reached_From : Node_Maps.Vector :=
        To_Vector (Ada.Containers.Count_Type (Node'Last - Node'First + 1));

**Comment by ARG-Editor** [comment]
I prefer splitting the baby: deleting the two paragraphs now (in AI12-0430-1), but deferring this issue to revisit it in the future (perhaps we should create a proper container-like iterator example somewhere, maybe 5.5.2, and if we did that we could use it here). I think we should delete the paragraphs now because incorrect examples can cause major confusion, particularly in this case when we're illustrating the use of iterators -- they really need to be properly declared so it doesn't appear that the compiler is conjuring them out of thin air.

**Comment by ARG-Editor** [comment]
Tucker's comment starting with "Interestingly" should also be deferred, since it is unclear whether there are any uses of Reached_From that expect being an array rather than a vector. (And it is an in an example that is not wrong and not modified, so it meets the standard reason for deferring - see #15.)
